cmd/infra/powervs/create.go

```diff
@@ -1234,43 +1211,38 @@ func (infra *Infra) isCloudConnectionReady(options *CreateInfraOptions, session
 	}
 
 	for retry := 0; retry < 2; retry++ {
-		f := func() (cond bool, err error) {
-			cloudConn, err = client.Get(infra.PowerVSCloudConnectionID)
+		f := func() (bool, error) {
+			cloudConn, err = client.Get(infra.CloudConnectionID)
 			if err != nil {
-				return
+				return false, err
 			}
 
 			if cloudConn != nil {
 				log(infra.ID).Info("Waiting for Cloud Connection to up", "id", cloudConn.CloudConnectionID, "status", cloudConn.LinkStatus)
 				if *cloudConn.LinkStatus == cloudConnectionEstablishedState {
-					cond = true
-					return
+					return true, nil
 				}
 			}
 
-			_, resp, err := directLinkV1.GetGatewayStatus(&directlinkv1.GetGatewayStatusOptions{ID: &infra.PowerVSCloudConnectionID, Type: &gatewayStatusType})
+			_, resp, err := directLinkV1.GetGatewayStatus(&directlinkv1.GetGatewayStatusOptions{ID: &infra.CloudConnectionID, Type: &gatewayStatusType})
 			if err != nil {
-				return
+				return false, err
 			}
 
 			log(infra.ID).Info("Status from Direct Link", "BGP", resp.Result)
 
-			return
+			return false, nil
 		}
 
-		err = wait.PollImmediate(pollingInterval, cloudConnEstablishedStateTimeout, f)
-		if err == nil {
+		if err = wait.PollImmediate(pollingInterval, cloudConnEstablishedStateTimeout, f); err == nil {
 			break
 		}
 	}
 	if cloudConn != nil {
 		infra.Stats.CloudConnState.Duration.Duration = time.Since(startTime)
 		infra.Stats.CloudConnState.Status = *cloudConn.LinkStatus
 	}
-	if err == nil {
-		log(infra.ID).Info("PowerVS Cloud Connection ready")
-		return
-	}
 
-	return
+	log(infra.ID).Info("PowerVS Cloud Connection ready")
```

Comment:
where are we handling the error case? Lets take a scenario where we exceeded the 3 retries for CC to come active state?